frontend/components/Settings/QueueTab.tsx

```diff
@@ -12,68 +13,205 @@ import {
   OBJECT_NAME_TITLE_SINGULAR,
   SITE_NAME,
 } from '../../utils/branding'
-import { Icon } from '../common'
+import { ModalContent } from '../ClubPage/Actions'
+import { Checkbox, Icon, Modal } from '../common'
+
+type QueueTableModalProps = {
+  show: boolean
+  closeModal: () => void
+  bulkAction: (comment: string) => void
+  isApproving: boolean
+}
+
+const QueueTableModal = ({
+  show,
+  closeModal,
+  bulkAction,
+  isApproving,
+}: QueueTableModalProps): ReactElement => {
+  const [comment, setComment] = useState<string>('')
+  return (
+    <Modal
+      show={show}
+      closeModal={() => {
+        setComment('')
+        closeModal()
+      }}
+      marginBottom={false}
+    >
+      <ModalContent>
+        <div className="mb-3">
+          Enter bulk {isApproving ? 'approval' : 'rejection'} notes here! Your
+          notes will be emailed to the requesters when you{' '}
+          {isApproving ? 'approve' : 'reject'} these requests.
+        </div>
+        <textarea
+          value={comment}
+          onChange={(e) => setComment(e.target.value)}
+          className="textarea my-2"
+          placeholder={`${isApproving ? 'approval' : 'rejection'} notes`}
+        ></textarea>
+        <button
+          className={`mb-2 button ${isApproving ? 'is-success' : 'is-danger'}`}
+          onClick={() => {
+            closeModal()
+            bulkAction(comment)
+          }}
+        >
+          <Icon name={isApproving ? 'check' : 'x'} />
+          {` ${isApproving ? 'Approve' : 'Reject'} and Send Message`}
+        </button>
+      </ModalContent>
+    </Modal>
+  )
+}
 
 type QueueTableProps = {
   clubs: Club[] | null
 }
-
+/* TODO: refactor with Table component when render and search
+functionality are disconnected */
 const QueueTable = ({ clubs }: QueueTableProps): ReactElement => {
-  if (clubs === null) {
-    return <div className="has-text-info">Loading table...</div>
+  const router = useRouter()
+  const [selectedCodes, setSelectedCodes] = useState<string[]>([])
+  const [showModal, setShowModal] = useState<boolean>(false)
+  const [approve, setApprove] = useState<boolean>(false)
+  const [loading, setLoading] = useState<boolean>(false)
+  const allClubsSelected = selectedCodes.length === (clubs || []).length
+
+  const toggleCode = (code: string) => {
+    if (selectedCodes.includes(code)) {
+      selectedCodes.splice(selectedCodes.indexOf(code), 1)
+    } else {
+      selectedCodes.push(code)
+    }
+    setSelectedCodes([...selectedCodes])
```

Comment:
this can be rewritten more immutable-friendly with:
```
const toggleCode = (code: string) => setSelectedCodes(selectedCodes.includes(code) ? selectedCodes.filter(c => c!== code) : [...selectedCodes, code])
```
The ternary operator might not be the best option readability-wise, so see if you can still take advantage of the spread operator!